### ISSUE TITLE
Windows 2016 Fails to Connect When Using Custom Credentials

### ISSUE THREAD

**Issue by jdgoins** [test-kitchen/kitchen-ec2#319]
When I provide custom credentials via my .kitchen.yml file, the ec2 driver is unable to connect to a new w2k16 instance with the supplied credentials. I am using w2k16 base latest (ami-c2c3a2a2 as of this writing). I have confirmed that the custom user is created and added to the Administrators group. This is not an issue on w2k12r2.

The run gets stuck on `[WinRM] opening remote shell on http://ec2-xx-xx-xx-xx.us-west-2.compute.amazonaws.com:5985/wsman`

The driver is able to connect using the Administrator account when I remove this portion from my .kitchen.yml file, transport section.
```
  username: chef
  password: '3rF5vBqnMEwmmWs4'
```

**Comment by grdnrio** [comment]
I also have this issue.

Looking at the code you can see the user-data being used to enable winrm access, but it doesn't seem to work on the latest 2016 AMI. I haven't tried running this manually, however. https://github.com/test-kitchen/kitchen-ec2/blob/master/lib/kitchen/driver/ec2.rb#L537-L585

It similar to what happens in the kitchen-azure driver: https://github.com/test-kitchen/kitchen-azurerm/blob/master/lib/kitchen/driver/azurerm.rb#L333-L379.

I'd assume that using the `user_data` property would override this, so it's probably a case of setting this for the 2016 platform. https://github.com/test-kitchen/kitchen-ec2#user_data

I'll share my user-data if I get it working.

**Comment by pschaumburg** [comment]
Can confirm the issue on my site.

**Comment by joshoohaah** [comment]
with w2k16 i am having just issues with kitchen-ec2 giving information back that the box is running.  In the AWS console it shows up and running. but in the debug output for test-kitchen it just keeps waiting for the box to become ready.
```
       Waited 890/900s for instance <instance_ID> to become ready.
D      Console output: ---
2017/07/26 16:00:17Z: Amazon SSM Agent v2.0.847.0 is running
```

**Comment by Sam-Martin** [comment]
I was having a similar issue, not sure if it would have worked had I removed the custom password. BUT I was able to resolve it by adding:

```
Set-ItemProperty -Name LocalAccountTokenFilterPolicy -Path HKLM:\software\Microsoft\Windows\CurrentVersion\Policies\system -Value 1
```
to the user_data. This should be done when `Enable-PSRemoting -Force -SkipNetworkProfileCheck` is run I believe, but that didn't seem to be happening for whatever reason.

My full `.kitchen.yml` looks like this:

```
---
driver:
  name: ec2
  region: eu-west-1
  security_group_ids: ["sg-aaaaaaaa"]
  require_chef_omnibus: 12
  instance_type: m3.medium
  associate_public_ip: true
  subnet_id: subnet-aaaaaaaa
  tags:
    Name: Test Kitchen test of default
verifier:
  name: inspec
transport:
  username: testkitchen
  password: P@Ssw0rd
provisioner:
  name: chef_zero
platforms:
  - name: windows-2016
    driver: 
      image_search:
        owner-id: 801119661308
        name: Windows_Server-2016-English-Full-Base-*
      user_data: |
        <powershell>
        $logfile="C:\ProgramData\Amazon\EC2-Windows\Launch\Log\kitchen-ec2.log"
        # Allow script execution
        Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Force
        #PS Remoting and & winrm.cmd basic config
        Enable-PSRemoting -Force -SkipNetworkProfileCheck
        & winrm.cmd set winrm/config '@{MaxTimeoutms="1800000"}' >> $logfile
        & winrm.cmd set winrm/config/winrs '@{MaxMemoryPerShellMB="1024"}' >> $logfile
        & winrm.cmd set winrm/config/winrs '@{MaxShellsPerUser="50"}' >> $logfile
        #Server settings - support username/password login
        & winrm.cmd set winrm/config/service/auth '@{Basic="true"}' >> $logfile
        & winrm.cmd set winrm/config/service '@{AllowUnencrypted="true"}' >> $logfile
        & winrm.cmd set winrm/config/winrs '@{MaxMemoryPerShellMB="1024"}' >> $logfile
        #Firewall Config
        & netsh advfirewall firewall set rule name="Windows Remote Management (HTTP-In)" profile=public protocol=tcp localport=5985 remoteip=localsubnet new remoteip=any  >> $logfile
        "Disabling Complex Passwords" >> $logfile
        $seccfg = [IO.Path]::GetTempFileName()
        & secedit.exe /export /cfg $seccfg >> $logfile
        (Get-Content $seccfg) | Foreach-Object {$_ -replace "PasswordComplexity\s*=\s*1", "PasswordComplexity = 0"} | Set-Content $seccfg
        & secedit.exe /configure /db $env:windir\security\new.sdb /cfg $seccfg /areas SECURITYPOLICY >> $logfile
        & cp $seccfg "c:\"
        & del $seccfg
        $username="testkitchen"
        $password="P@Ssw0rd"
        "Creating static user: $username" >> $logfile
        & net.exe user /y /add $username $password >> $logfile
        "Adding $username to Administrators" >> $logfile
        & net.exe localgroup Administrators /add $username >> $logfile
        Set-ItemProperty -Name LocalAccountTokenFilterPolicy -Path HKLM:\software\Microsoft\Windows\CurrentVersion\Policies\system -Value 1
        </powershell>


suites:
  - name: default
    run_list:
      - recipe[default::default]
    attributes:

```

**Comment by cheeseplus** [comment]
Fixed by #348
